### DISCUSSION TITLE
Does training/processing implicitly happen at the sentence level?

### DISCUSSION THREAD

**Discussion by Zatteliet** [explosion/spaCy#10660] [Help: Other Questions] [answered]
I cannot easily find a confirm/deny on this. Let's say I make a `Doc` from a text with multiple sentences and apply `ner` on it (or any other component). Does the pipeline implicitly sentencize the input and process it sentence by sentence?

In other words, could I potentially get different results if I sentencize the input first and feed it to the pipeline piece by piece?

Does it work the same way during training?

**Comment by polm** [comment, marked as answer]
The pipeline just passes Docs to each component - at the global level there's no special treatment of sentences. (That's the reason this isn't "documented" - there's nothing special to document.)

Within a single component sentence information on a Doc is visible and can be used, though how it's used differs by component. For example, the NER component won't mark an entity across sentence boundaries. Behavior like this may be subtle, but it should be unsurprising.

> In other words, could I potentially get different results if I sentencize the input first and feed it to the pipeline piece by piece?

Yes, they can be different. It depends on the component but at a minimum you'll get different contexts for your tokens at sentence boundaries, which can affect many components. I would not expect the differences to be drastic.

**Reply by Zatteliet** [reply]
That's clear, thank you.
